NBitcoin.Altcoins/Elements/ElementsTransaction.cs

```diff
@@ -603,9 +606,20 @@ protected override ConfidentialAsset GetAssetCore()
 
 	public class ElementsTransaction<TNetwork> : ElementsTransaction
 	{
+		private ElementsConsensusFactory<TNetwork> _consensusFactory;
+
+		public ElementsTransaction()
+		{
+
+		}
+
+		public ElementsTransaction(ElementsConsensusFactory<TNetwork> consensusFactory)
+		{
+			_consensusFactory = consensusFactory;
+		}
 		public override ConsensusFactory GetConsensusFactory()
 		{
-			return ElementsConsensusFactory<TNetwork>.Instance;
+			return _consensusFactory;
```

Comment:
I worked hard so we don't have to do this.
Why can't we do what was done before?

Reply:
done